evcxr/src/eval_context.rs

```diff
@@ -219,12 +220,22 @@ impl EvalContext {
                     syn::Stmt::Item(syn::Item::ExternCrate(syn::ItemExternCrate {
                         ident, ..
                     })) => {
-                        let crate_name = ident.to_string();
+                        let mut crate_name = ident.to_string();
                         if !self.dependency_lib_names()?.contains(&crate_name) {
                             self.state
                                 .external_deps
                                 .entry(crate_name.clone())
                                 .or_insert_with(|| {
+                                    // For RFC0940
+                                    if crate_name.contains("_") {
+                                        let crates_client = SyncClient
+                                            ::new("evcxr", std::time::Duration::from_millis(500)).unwrap();
```

Comment:
`cargo-edit` has an implementation of this that uses the local registry instead of the crates.io api: [`fuzzy_query_registry_index`](https://github.com/killercup/cargo-edit/blob/f7c89623813023a604fad50512a5dc12cd985548/src/fetch.rs#L263-L297)